app/js/components/tabs/tabcomponents/observationComponent.jsx

```diff
@@ -1,165 +1,381 @@
-        this.getDateString = this.getDateString.bind(this);
-    }
-    componentDidMount(){}
-    getFormData(fields) {
-      const searchParams = {};
-      for(const eachField in fields) {
-        if(Array.isArray(fields[eachField])) {
-            fields[eachField].forEach((fieldInput, index) => {
-                let fieldValue = $('#obsForm').find('#'+fieldInput.name).val();
-                fields[eachField][index].value = (fields[eachField][index].isArrayValue && fieldValue != '')
-                    ? [ fieldValue ]
-                    : fieldValue;
-            });
-            searchParams[eachField] = fields[eachField];
-            continue;
-        }
-      }
-      return searchParams;
-    }
-    searchObservation(e) {
-        e.preventDefault();
-        const startDate = this.state.startDate;
-        const endDate = this.state.endDate;
-        const timeModifier = document.getElementById('timeModifier').value;
-        let historyTitle = (timeModifier === "ANY")
-            ? `Patients with ${this.props.conceptName.name}`
-            : `Patients without ${this.props.conceptName.name}`;
-        const fields = {
-            codedObsSearchAdvanced : [
-                { name: 'question'},
-                { name: 'timeModifier'}
-            ]
-        };
-        if(startDate !== "") {
-            historyTitle += ` since ${startDate}`;
-            fields.codedObsSearchAdvanced.push({ name: 'startDate'});
-        }
-        if(endDate !== "") {
-            historyTitle += ` until ${endDate}`;
-            fields.codedObsSearchAdvanced.push({ name: 'endDate'});
-        }
+        this.handleFormChange = this.handleFormChange.bind(this);
+        this.nullDatatype = this.nullDatatype.bind(this);
+    }
 
-        const searchParams = this.getFormData(fields);
+    componentDidUpdate() {
+        this.textInput ?  this.textInput.focus() : false;
+    }
+
+    handleFormChange(event) {
+        event.preventDefault();
         this.setState({
-            searchResults: searchParams
-        });
-        const searchData = this.jsonHelper.composeJson(searchParams);
-        this.props.search(searchData).then(results => {
-            const obsType = results.rows || [];
-            this.props.addToHistory(historyTitle, obsType, results.query);
+            [ event.target.name ] :  event.target.value
         });
     }
 
-    /**
-     * Method to update the date key for different date types in the state
-     * @param {String} stateKey - The key in the component state that should be
-     * updated
-     * @return {Function} - Call back function to be executed by the date input
-     * field
-     */
-    handleDateChange(dateType) {
-        return value => this.setState({
-            [dateType]: this.getDateString(value)
+    setRef(input) {
+        this.textInput = input; 
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+        const types = {
```

Comment:
Link to the observation component [click here](https://github.com/andela-odaniel/openmrs-owa-cohortbuilder/blob/118097f6aa05966e72d0164b53e58a3b8c2f95b1/app/js/components/tabs/tabcomponents/observationComponent.jsx)

![screen shot 2017-07-03 at 9 53 58 pm](https://user-images.githubusercontent.com/23451291/27807642-3630945e-603a-11e7-94d8-6ac32d96a4fd.png)